### PR TITLE
feat: upgrade react native to version 0.72.5

### PR DESCRIPTION
**Pull request by samshara**

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -33,7 +33,7 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath("com.android.tools.build:gradle:7.3.1")
+        classpath("com.android.tools.build:gradle")
```
Comment (laurentS) [inline android/build.gradle:36]: Shouldn't this version still be pinned?

**Review by laurentS — COMMENTED** [review]
Looks ok to me, pending CI tests.

File: android/build.gradle (hunk 1 above)
Comment (samshara) [reply android/build.gradle:36]: @laurentS I used the update helper tool. It suggest the changes here
https://react-native-community.github.io/upgrade-helper/?from=0.71.3&to=0.72.5#RnDiffApp-android-build.gradle

**Comment by kopitek8** [comment]
Hey @laurentS. The upgrade is complete from our end pending pull request review. Please have a look at it and review it from your side. 

Looping @samshara here as well

**Review by laurentS — APPROVED** [review]
It's as difficult to review these RN upgrades as it is to write them :(
I can't really check the `project.pbxproj` file, it's really just garbage without access to a mac.
As written in the comment, my only concern is about the version code calculation. If that works, then all good.
There are a number of warnings and deprecation notices in the CI that are probably worth taking care of sooner rather than later, seeing how the stores love blocking anything they can. In separate PRs though :)
Thanks for taking care of these upgrades! :raised_hands:

File: android/app/build.gradle [hunk 2]
```diff
@@ -195,30 +167,12 @@ android {
...
-    applicationVariants.all { variant ->
-        variant.outputs.each { output ->
-            // For each separate APK per architecture, set a unique version code as described here:
-            // https://developer.android.com/studio/build/configure-apk-splits.html
-            // Example: versionCode 1 will generate 1001 for armeabi-v7a, 1002 for x86, etc.
-            def versionCodes = ["armeabi-v7a":1, "x86":2, "arm64-v8a": 3, "x86_64": 4]
-            def abi = output.getFilter(OutputFile.ABI)
-            if (abi != null) {  // null for the universal-debug, universal-release variants
-                output.versionCodeOverride =
-                        defaultConfig.versionCode * 1000 + versionCodes.get(abi)
```
[the remark this answers is not in the document]
Comment (samshara) [reply android/app/build.gradle:209]: @laurentS I will have a look at it.

[the remark this answers is not in the document]
Comment (samshara) [reply android/app/build.gradle:209]: @laurentS please have a look at the changes suggested by the upgrade helper tool. I suppose this part of the file was not modified and we can make the suggested changes. 
https://react-native-community.github.io/upgrade-helper/?from=0.71.3&to=0.72.5#RnDiffApp-android-app-build.gradle

[the remark this answers is not in the document]
Comment (samshara) [reply android/app/build.gradle:209]: @laurentS We couldn't access the [playstore](https://play.google.com/console/u/0/developers/7751656606363587225/app/4975116387986139576/tracks/internal-testing) link to verify. We will deploy to the test server, and look at the version numbers and verify if they are equal.

[the remark this answers is not in the document]
Comment (laurentS) [reply android/app/build.gradle:209]: Hi @samshara I don't really have time to dig into any of this but:
- the run error says `Your certificate '43JBSHZNKH.cer' is not valid, please check end date and renew it if necessary` you need to renew the apple certs. See https://github.com/mapswipe/mapswipe/pull/572 and maybe https://github.com/mapswipe/mapswipe/pull/344/files (maybe outdated)
- for the version number of the app, be careful that the stores only ever accept strictly monotonic sequences of numbers. once you upload a version N, you can only ever upload bigger numbers afterwards, if I remember correctly.

Good luck. These bits are annoying to get right.

[the remark this answers is not in the document]
Comment (samshara) [reply android/app/build.gradle:209]: Thanks Laurent, for the clarification on the issues. I will have a look at the version numbers and verify them as well.

File: android/build.gradle (hunk 1 above)
Comment (samshara) [reply android/build.gradle:36]: @laurentS  can you add me to the `ios-certificates` Gitlab repository? 
My gitlab username is `@samshara`.

File: android/app/build.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (samshara) [reply android/app/build.gradle:209]: @laurentS  can you add me to the `ios-certificates` Gitlab repository? 
My gitlab username is `@samshara`.

[the remark this answers is not in the document]
Comment (laurentS) [reply android/app/build.gradle:209]: @samshara done! I had to remove one member as we seem to have a 5 user limit on the repo now :(
Let us know if anything is not working as expected.
Just a note, I remember that at some point, you have to delete certificates and keys from the repo. I was scared of doing something wrong the first time, but it's actually expected.

**Pull request merged**